llvm/lib/CodeGen/GlobalISel/LegalizerHelper.cpp

```diff
@@ -5496,6 +5496,26 @@ LegalizerHelper::moreElementsVector(MachineInstr &MI, unsigned TypeIdx,
     return Legalized;
   }
 
+  case TargetOpcode::G_SEXT:
```

Comment:
There is a case above that handles G_TRUNC and a few others that have different type sizes. It could be extended to handle these operations too.

Reply:
Awesome, that works - though I have to set AArch64LegalizerInfo to use index 0 instead of 1.